src/depackers/vorbis.c

```diff
@@ -1115,10 +1115,7 @@ static int compute_codewords(Codebook *c, uint8 *len, int n, uint32 *values)
    memset(available, 0, sizeof(available));
    // find the first entry
    for (k=0; k < n; ++k) if (len[k] < NO_CODE) break;
-
-   // sanity check
-   if (k == n) return (c->sorted_entries == 0);
-
+   if (k == n) { assert(c->sorted_entries == 0); return TRUE; }
```

Comment:
The variable of interest here is `c->sorted_entries`. `compute_codewords` is called by `start_decoder`. (line 3886)

## Case 1: ordered
1) `sorted_count` = 0 (line 3856)
2) The following loop scans for values != `NO_CODE` of a certain length; `sorted_count++` when found
3) `c->sorted_entries` = `sorted_count` (line 3864)
4) `c->sorted_entries` will always be 0 if all codes are `NO_CODE`, so `compute_codewords` returns `TRUE`. (line 1120)
Also, it seems like there isn't even a mechanism for `NO_CODE` to get inserted in this case, which makes sense.

## Case 2: unordered, non-sparse.
1) `total` incremented only if a code exists, not when `NO_CODE` is inserted (line 3834)
2) Because `total` is unused, this case ends up being identical to case 1.

## Case 3: unordered, sparse.
1) `total` = 0 (line 3787)
2) `total` incremented only if a code exists, not when `NO_CODE` is inserted (line 3834)
3) side case: sparse can be converted to non-sparse, which will follow the same path as case 1. (line 3839-3850)
4) `sorted_count` = `total` (line 3854)
5) `c->sorted_entries` = `sorted_count` (line 3864)
6) `c->sorted_entries` will always be 0 if all codes are `NO_CODE`, so `compute_codewords` returns `TRUE`. (line 1120)